kubernetes-client/pom.xml

```diff
@@ -107,6 +103,12 @@
             <groupId>io.fabric8</groupId>
             <artifactId>kubernetes-model-node</artifactId>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.osgi/org.osgi.service.component.annotations -->
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.service.component.annotations</artifactId>
+            <version>1.5.1</version>
+        </dependency>
```

Comment:
Do we _really_ need it? Can we instead exclude the OSGi specific parts from transitive dependencies or filter out from the sources? At minimum remove the comment?

Reply:
I was expecting you're getting compilation errors. But look closer. It seems that it's just a single class `io.fabric8.kubernetes.client.osgi.ManagedKubernetesClient` that is a problem here. And it seems to be OSGi specific. We're _not_ on OSGi any  more, so we're most likely _not_ using this class. Hence, instead of adding a dependency that will require compliance in the next step, we can remove this single class.